### DISCUSSION TITLE
Parametric Round Adapter

### DISCUSSION THREAD

**Discussion by DavidCC00** [kennetek/gridfinity-rebuilt-openscad#61] [Ideas]
1.  This option would switch from the normal bin to a solid bin with a round hole, allowing easy creation of round sockets for all those cylinders in our lives.
2.  Extra Fancy:  A repeating pattern for multiple round objects in one bin.
3.  An option for a round lip inset for the circle part, of the same dimensions of the normal stacking lip.  
4.  A round bottom "bin socket" option to compliment the above, allowing for the creation of round risers/lids/inserts.

Inspiration for this idea:  Turning Factorio/Geodesic Dome this into a gridfinity lightshade.

https://www.thingiverse.com/thing:4678578

Similar to this:

https://www.printables.com/model/379261-led-tea-light-lantern-lamp

But also for gluesticks and such.

**Comment by Ruudjhuu** [comment]
For point 1 and 2, you mean something like this: 

![image](https://user-images.githubusercontent.com/31508944/216453266-d4798c97-b0c6-4ec1-9b91-8430f27da10d.png)

It is very easy to do, numbers are just random:
```c
color("tomato") {
  gridfinityInit(gridx, gridy, height(gridz, gridz_define, enable_lip, enable_zsnap), height_internal, length) {
      pattern_linear(2,3,18,10)
      cylinder(100,7,true);
  }
  gridfinityBase(gridx, gridy, length, div_base_x, div_base_y, style_hole);
}
```

regarding point 3 and 4, I do not fully understand. I do not see how the thingyverse link is applicable to gridfinity.

The printables link is very unique use-case. The bottom parts can be created with the code snippet above. The shades requires to be written from scratch with only this use-case.

**Reply by DavidCC00** [reply]
1 & 2.  Thanks!    Am I mistaken, or is "slit" only with screw holes?  I don't like printing the screw holes, as I don't use them and they worsen the corner warping with certain filaments.   As for the extra layers/squares, that's in case the person's printer's cooling isn't very good or they're using exotic filament/settings like CNCKitchen's guide for printing PETG/PCTG strong and transparent (minimum cooling, hottest temp possible).   For example, my first printer (Artillery Sidewinder X1) had terrible cooling until I printed an adapter for a larger blower and replaced the bushing fan with a Sunon maglev. 

3 & 4.  I was also envisioning round covers and risers.  For example, similar to these, but round:

risers/spacers  https://youtu.be/Bd4NnHvTRAY

https://www.printables.com/model/387051-xxl-dust-cover-for-gridfinity-ts100-all-in-one-org

Round is a bit stronger (especially in vase mode), a bit faster/less filament, and is easier to grab in tightly packed spaces (the corners become natural finger holes).  

I might be wrong, maybe it's more niche, but personally for the things I haven't found but want to make, there's a few.    I could do it in Tinkercad (I have both F360 and SCAD but still very much a noob), but Tinkercad has all kinds of issues with the models it makes, and isn't parametric at all. 

Of course, it's your decision, since I'm still a novice in this area. 

Thank you!

**Comment by kennetek** [comment]
The linear pattern for cylinders may be good to implement as a separate file. It seems to be something people want, and can easily be used for a variety of items. 

As for the other ideas, are you suggesting round gridfinity (or round covers...)? All the links in this discussion are somewhat scattered in terms of what they accomplish.

**Reply by DavidCC00** [reply]
I was thinking round covers, and round "lips" for those covers to sit in.  

The "scattered" was me trying to show there's a variety of applications I've wanted to use this in.  I've actually thought of a couple more, but I don't want to sideline this anymore than I already have, lol.
